### PR TITLE
 Ignoring changes to allow admins to un-enroll users from MFA.

### PR DESCRIPTION
**Pull request by quinteroj**
Reverting changes regarding MFA users un-enrollment by admin.

### CODE REVIEW

**Comment by quinteroj** [comment]
Keep the code to force users to enroll in MFA if the global MFA setting is turned ON.

**Comment by lmajano** [comment]
@quinteroj can you create a ticket in the ContntBox project for this so we can track it.

**Comment by lmajano** [comment]
https://ortussolutions.atlassian.net/secure/RapidBoard.jspa?rapidView=21&projectKey=CONTENTBOX&selectedIssue=CONTENTBOX-978

File: modules/contentbox/modules/contentbox-admin/handlers/authors.cfc [hunk 1]
```diff
@@ -332,7 +338,7 @@ component extends="baseHandler"{
         flash.put( "authorID", rc.authorID );
         prc.oAuthor = authorService.get( id=rc.authorID );
 
-        if ( prc.oCurrentAuthor.getAuthorID() != prc.oAuthor.getAuthorID() ) {
+        if ( prc.oCurrentAuthor.isLoaded() && prc.oCurrentAuthor.getAuthorID() != prc.oAuthor.getAuthorID() ) {
```
Comment (lmajano) [inline modules/contentbox/modules/contentbox-admin/handlers/authors.cfc]: why did you add the `isLoaded()`?

File: modules/contentbox/modules/contentbox-admin/handlers/authors.cfc [hunk 2]
```diff
@@ -354,7 +360,7 @@ component extends="baseHandler"{
         var vResults = validateModel( target = prc.oAuthor, excludes = "password" );
 		if ( vResults.hasErrors() ) {
             cbMessagebox.warn( messageArray=vResults.getAllErrors() );
-            flash.keep( "layout,xehInvalidData,xehEnrollmentSuccess,xehEnrollmentSuccessQueryString" );
+            flash.keep( "layout,xehInvalidData,xehEnrollmentSuccess,xehEnrollmentSuccessQueryString" )
```
Comment (lmajano) [inline modules/contentbox/modules/contentbox-admin/handlers/authors.cfc]: At this point, just do a `flash.keep()` with not items, this will keep all flashed state.

File: modules/contentbox/modules/contentbox-admin/modules/contentbox-security/handlers/security.cfc [hunk 3]
```diff
@@ -75,6 +75,16 @@ component extends="baseHandler"{
 				);
 			}
 
+			//If Global MFA is turned on and the user is not enrolled to a provider, then force it to enroll
```
Comment (lmajano) [inline modules/contentbox/modules/contentbox-admin/modules/contentbox-security/handlers/security.cfc]: Spacing betwee `// If`

File: modules/contentbox/modules/contentbox-admin/views/twofactor/forceEnrollment.cfm [hunk 4]
```diff
@@ -79,7 +79,7 @@
                         Action Bar
                         Saving only if you have permissions, else it is view only.
                     --->
-                    <cfif prc.oCurrentAuthor.checkPermission( "AUTHOR_ADMIN" ) OR prc.author.getAuthorID() EQ prc.oCurrentAuthor.getAuthorID()>
+                    <cfif ( !prc.oCurrentAuthor.isLoaded() AND prc.author.isLoaded() ) OR ( prc.author.getAuthorID() EQ prc.oCurrentAuthor.getAuthorID() OR prc.oCurrentAuthor.checkPermission( "AUTHOR_ADMIN" ) )>
```
Comment (lmajano) [inline modules/contentbox/modules/contentbox-admin/views/twofactor/forceEnrollment.cfm]: Why do an is loaded check? You check if not loaded or loaded? Doesn't make sense

File: modules/contentbox/modules/contentbox-admin/handlers/authors.cfc [hunk 5]
```diff
@@ -438,6 +444,10 @@ component extends="baseHandler"{
 		authorService.saveAuthor( oAuthor );
 		// message
 		cbMessagebox.setMessage( "info","Two Factor Settings Saved!" );
+		if ( !prc.oCurrentAuthor.isLoaded() ){
```
Comment (lmajano) [inline modules/contentbox/modules/contentbox-admin/handlers/authors.cfc]: I am not sure why you are doing this? What is the point of this?

**Comment by quinteroj** [comment]
@lmajano Done
https://ortussolutions.atlassian.net/browse/CONTENTBOX-984

File: modules/contentbox/modules/contentbox-admin/views/twofactor/forceEnrollment.cfm (hunk 4 above)
Comment (quinteroj) [reply modules/contentbox/modules/contentbox-admin/views/twofactor/forceEnrollment.cfm]: I added the isLoaded check because at that point the user is not logged in but it is authenticated, so If the user is valid (loaded) I should force to enroll in MFA.

File: modules/contentbox/modules/contentbox-admin/handlers/authors.cfc (hunk 1 above)
Comment (quinteroj) [reply modules/contentbox/modules/contentbox-admin/handlers/authors.cfc]: isLoaded check tells us if a user is authenticated but not logged in. We need that flag to know if the user is who is trying to log in or if this user is an admin who is already logged in. We can discuss the strategy if this is not very clean.

File: modules/contentbox/modules/contentbox-admin/handlers/authors.cfc (hunk 2 above)
Comment (quinteroj) [reply modules/contentbox/modules/contentbox-admin/handlers/authors.cfc]: done

File: modules/contentbox/modules/contentbox-admin/handlers/authors.cfc (hunk 5 above)
Comment (quinteroj) [reply modules/contentbox/modules/contentbox-admin/handlers/authors.cfc]: Not sure about this. That code was already there, I am afraid of taking it out. It might mess up something that is using the author stored in the session.

File: modules/contentbox/modules/contentbox-admin/modules/contentbox-security/handlers/security.cfc (hunk 3 above)
Comment (quinteroj) [reply modules/contentbox/modules/contentbox-admin/modules/contentbox-security/handlers/security.cfc]: Fixed

File: modules/contentbox/modules/contentbox-admin/handlers/authors.cfc [hunk 6]
```diff
@@ -311,7 +311,13 @@ component extends="baseHandler"{
      * Show the two-factor authentication screen for forced enrollment
      */
     function forceTwoFactorEnrollment( event, rc, prc ) {
-        prc.author = prc.oCurrentAuthor;
+		if( flash.exists( "authorData" ) ){
+			// Inflate author for requested events
+			prc.author = authorService.get( flash.get( "authorData" ).authorID );
```
Comment (lmajano) [inline modules/contentbox/modules/contentbox-admin/handlers/authors.cfc]: @gpickin is this the  risk that you were saying?

File: modules/contentbox/modules/contentbox-admin/handlers/authors.cfc (hunk 1 above)
Comment (lmajano) [reply modules/contentbox/modules/contentbox-admin/handlers/authors.cfc]: But `getAuthorID()` will be empty, so no need for a second test. Did you test with an invalid user id?

File: modules/contentbox/modules/contentbox-admin/handlers/authors.cfc (hunk 5 above)
Comment (lmajano) [reply modules/contentbox/modules/contentbox-admin/handlers/authors.cfc]: Let me rephrase, why log in the user?

File: modules/contentbox/modules/contentbox-admin/modules/contentbox-security/handlers/security.cfc [hunk 7]
```diff
@@ -75,6 +75,16 @@ component extends="baseHandler"{
...
 			}
 
+			// If Global MFA is turned on and the user is not enrolled to a provider, then force it to enroll
+			if( twoFactorService.isForceTwoFactorAuth() AND !results.author.getIs2FactorAuth() ){
+				flash.put( "authorData", {
+					authorID 	= results.author.getAuthorID(),
+					rememberMe 	= rc.rememberMe,
+					securedURL  = rc._securedURL
+				} );
+				setNextEvent( "#prc.cbAdminEntryPoint#.authors.forceTwoFactorEnrollment" );
```
Comment (lmajano) [inline modules/contentbox/modules/contentbox-admin/modules/contentbox-security/handlers/security.cfc]: I think this is the issue @gpickin . You should not force the enrollment by going to a secured author handler. The enforcement should stay within the two factor module. This cannot be.

File: modules/contentbox/modules/contentbox-admin/handlers/authors.cfc (hunk 6 above)
Comment (gpickin) [reply modules/contentbox/modules/contentbox-admin/handlers/authors.cfc]: No because this is only loading if we see it in the flash.
There is another part of the flow that just looks for the authorid

moving it should fix that issue, i think.

File: modules/contentbox/modules/contentbox-admin/modules/contentbox-security/handlers/security.cfc (hunk 7 above)
Comment (gpickin) [reply modules/contentbox/modules/contentbox-admin/modules/contentbox-security/handlers/security.cfc]: k

**Comment by lmajano** [comment]
Merged manually, please see latest commit
